helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -573,9 +573,17 @@ public void resetPartition(String clusterName, String instanceName, String resou
     // check the instance is alive
     LiveInstance liveInstance = accessor.getProperty(keyBuilder.liveInstance(instanceName));
     if (liveInstance == null) {
-      throw new HelixException(
-          "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
-              + ", because " + instanceName + " is not alive");
+      // check if the instance has ever exist in the cluster
+      String instancePath = PropertyPathBuilder.instance(clusterName, instanceName);
+      if (!_zkClient.exists(instancePath)) {
+        throw new HelixException(
+            "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
+            + ", because " + instanceName + " never existed in cluster " + clusterName);
+      } else {
+        throw new HelixException(
+            "Can't reset state for " + resourceName + "/" + partitionNames + " on " + instanceName
+            + ", because " + instanceName + " is not alive in cluster " + clusterName + " anymore");
```

Comment:
Can you just create the error message in the if-else block and throw only once to avoid code duplication?

Reply:
TFTR. Will fix.